Source/Dafny/Dafny.atg

```diff
@@ -705,16 +704,15 @@ TOKENS
   digits = digit {['_'] digit}.
   hexdigits = "0x" hexdigit {['_'] hexdigit}.
   decimaldigits = digit {['_'] digit} '.' digit {['_'] digit}.
-  arrayToken = "array" [('1' digit | posDigitFrom2 ) {digit}].
-  arrayToken_q = "array" [('1' digit | posDigitFrom2 ) {digit}] '?'.
+  arrayToken = "array" [('1' digit | posDigitFrom2 ) {digit}] ['?'].
   bvToken = "bv" ( '0' | posDigit {digit} ).
   bool = "bool".
   char = "char".
   int = "int".
   nat = "nat".
   real = "real".
   ORDINAL = "ORDINAL".
-  object = "object".
+  object = "object". // Keeping object and object? as literals simplifies ident
   object_q = "object?".
```

Comment:
Can `object` and `object_q` be combined in the same way as `arrayToken` and `arrayToken_q` were combined?

Reply:
I tried, but then they also have to be parsed the way arrayToken and bvToken are in the ident production (L.686). I considered it not worth the effort and complication.